src/igraph/drawing/matplotlib/graph.py

```diff
@@ -26,214 +27,261 @@
+    Artist = mpl.artist.Artist
+except AttributeError:
+    Artist = FakeModule
 
 #####################################################################
 
 
-class MatplotlibGraphDrawer(AbstractGraphDrawer):
-    """Graph drawer that uses a pyplot.Axes as context"""
-
-    _shape_dict = {
-        "rectangle": "s",
-        "circle": "o",
-        "hidden": "none",
-        "triangle-up": "^",
-        "triangle-down": "v",
-    }
+class _Unset:
+    def __repr__(self):
+        return "<UNSET>"
+
+
+_UNSET = _Unset()
+
+
+# NOTE: https://github.com/networkx/grave/blob/main/grave/grave.py
+def _stale_wrapper(func):
+    """Decorator to manage artist state."""
+
+    @wraps(func)
+    def inner(self, *args, **kwargs):
+        try:
+            func(self, *args, **kwargs)
+        finally:
+            self.stale = False
+
+    return inner
+
+
+def _forwarder(forwards, cls=None):
+    """Decorator to forward specific methods to Artist children."""
+    if cls is None:
+        return partial(_forwarder, forwards)
+
+    def make_forward(name):
+        def method(self, *args, **kwargs):
+            ret = getattr(cls.mro()[1], name)(self, *args, **kwargs)
+            for c in self.get_children():
+                getattr(c, name)(*args, **kwargs)
+            return ret
+
+        return method
+
+    for f in forwards:
+        method = make_forward(f)
+        method.__name__ = f
+        method.__doc__ = "broadcasts {} to children".format(f)
+        setattr(cls, f, method)
+
+    return cls
+
+
+def _additional_set_methods(attributes, cls=None):
+    """Decorator to add specific set methods for children properties."""
+    if cls is None:
+        return partial(_additional_set_methods, attributes)
+
+    def make_setter(name):
+        def method(self, value):
+            self.set(**{name: value})
+        return method
+
+    for attr in attributes:
+        desc = attr.replace('_', ' ')
+        method = make_setter(attr)
+        method.__name__ = f"set_{attr}"
+        method.__doc__ = f"Set {desc}."
+        setattr(cls, f"set_{attr}", method)
+
+    return cls
+
+
+@_additional_set_methods(
+    (
+        "vertex_color",
+        "vertex_size",
+        "vertex_font",
+        "vertex_label",
+        "vertex_label_angle",
+        "vertex_label_color",
+        "vertex_label_dist",
+        "vertex_label_size",
+        "vertex_order",
+        "vertex_shape",
+        "vertex_size",
+        "edge_color",
+        "edge_curved",
+        "edge_font",
+        "edge_arrow_size",
+        "edge_arrow_width",
+        "edge_width",
+        "edge_label",
+        "edge_background",
+        "edge_align_label",
+        "autocurve",
+        "layout",
+    )
+)
+@_forwarder(
+    (
+        "set_clip_path",
+        "set_clip_box",
+        "set_transform",
+        "set_snap",
+        "set_sketch_params",
+        "set_figure",
+        "set_animated",
+        "set_picker",
+    )
+)
+class Graph(Artist, AbstractGraphDrawer):
```

Comment:
Can we rename this class to `GraphArtist` instead to avoid confusion with the `Graph` class (especially when this class starts to appear in the index of the API documentation, right next to the "real" `Graph` class).

Reply:
I think we also need to rename the reference to this class from the tutorial if we decide to do the renaming.